### PR TITLE
Mile3

### PR DESCRIPTION
**Pull request by paulzay**
implement interactive form 
add books to state
remove books to state

### CODE REVIEW

File: src/containers/BookList/BookList.js [hunk 1]
```diff
@@ -17,19 +17,21 @@ function BookList({ books }) {
...
           ))
         }
       </tbody>
     </table>
   );
 }
 
+BookList.propTypes = {
+  books: PropTypes.instanceOf(Array).isRequired,
+  deleteBook: PropTypes.func.isRequired,
```
Comment (mosaaleb) [inline src/containers/BookList/BookList.js:30]: - [ ] I got this error console:
Screenshot from 2020-12-03 19-02-10

We still missing deleteBook function in this component.

File: src/components/Book/Book.js [hunk 2]
```diff
@@ -1,22 +1,33 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-
+import { connect } from 'react-redux';
```
Comment (mosaaleb) [inline src/components/Book/Book.js]: - [ ] Book component should still remain presentational component. It should not connect to the store. As the prevoius comment, deleteBook should be passed down from `BookList` component as it is a container component (connects to redux store).
Please take a look at this [article](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0)

File: src/components/Book/Book.js [hunk 3]
```diff
@@ -1,22 +1,33 @@
...
+  book: PropTypes.shape({
+    id: PropTypes.number.isRequired,
+    title: PropTypes.string.isRequired,
+    category: PropTypes.string.isRequired,
+  }).isRequired,
+  deleteBook: PropTypes.func.isRequired,
+};
 
-export default Book;
+const mapDispatchToProps = dispatch => ({
```
Comment (mosaaleb) [inline src/components/Book/Book.js]: - [ ] This should be moved to `BookList` Component, and pass down `deleteBook` to every `Book`

**Review by mosaaleb — CHANGES_REQUESTED** [review]
### Changes Required 

Well done @paulzay  :clap:. You have perfectly handled most of the requirements. Still, we have some requirements to handle before moving on. Take a look at my comments below.

Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

**Review by lucasmazo23 — APPROVED** [review]
## Project approved 🚀
### Great job!
### Your project looks great!
### You can go ahead and merge this pull request!
### Happy coding! 💻
## Lucas Mazo

**Pull request merged**